ehrapy/preprocessing/_data_imputation.py

```diff
@@ -386,6 +386,283 @@ def miss_forest_impute(
     return adata
 
 
+# ===================== SoftImpute =========================
+
+
+def soft_impute(
+    adata: AnnData,
+    var_names: list[str] | None = None,
+    copy: bool = False,
+    warning_threshold: int = 30,
+    shrinkage_value: float | None = None,
+    convergence_threshold: float = 0.001,
+    max_iters: int = 100,
+    max_rank: int | None = None,
+    n_power_iterations: int = 1,
+    init_fill_method: str = "zero",
+    min_value: float | None = None,
+    max_value: float | None = None,
+    normalizer: object | None = None,
+    verbose: bool = True,
+) -> AnnData:
+    """Impute data using the SoftImpute.
+
+    See https://github.com/iskandr/fancyimpute/blob/master/fancyimpute/soft_impute.py
+    Matrix completion by iterative soft thresholding of SVD decompositions.
+
+    Args:
+        adata: The AnnData object to use SoftImpute on.
+        var_names: A list of var names indicating which columns to impute (if None -> all columns).
+        copy: Whether to return a copy or act in place.
+        warning_threshold: Threshold of percentage of missing values to display a warning for (default: 30).
+        shrinkage_value : Value by which we shrink singular values on each iteration. If omitted then the default value will be the maximum singular value of the initialized matrix (zeros for missing values) divided by 50.
+        convergence_threshold : Minimum ration difference between iterations (as a fraction of the Frobenius norm of the current solution) before stopping.
+        max_iters: Maximum number of SVD iterations.
+        max_rank: Perform a truncated SVD on each iteration with this value as its rank.
+        n_power_iterations: Number of power iterations to perform with randomized SVD.
+        init_fill_method: How to initialize missing values of data matrix, default is to fill them with zeros.
+        min_value: Smallest allowable value in the solution.
+        max_value: Largest allowable value in the solution.
+        normalizer: Any object (such as BiScaler) with fit() and transform() methods.
+        verbose: Print debugging info.
+
+    Returns:
+        The imputed (but unencoded) AnnData object
```

Comment:
Thanks! I have a small question: I just copied this from the previous ones (miss_forest_impute and knn_impute), does this "unencoded" mean has not encoded to one-hot encoding?

Reply:
Yeah I should remove those there as well. Encoding happensw during Imputation but thats an implementation detail the user does not have to be aware of, since itself completely reverted in the end. 
Guess the `unencoded` here would distract user more than it'd help.